src/containerapp/azext_containerapp/containerapp_sessionpool_decorator.py

```diff
@@ -483,11 +522,27 @@ def set_up_managed_identity_settings(self):
             safe_set(self.session_pool_def, "properties", "managedIdentitySettings", value=managed_identity_settings)
 
     def set_up_dynamic_configuration(self):
+        lifecycle_config_def = {}
+
+        if self.get_argument_max_alive_period() is not None and self.get_argument_lifecycle_type() is not None and self.get_argument_lifecycle_type().lower() != "oncontainerexit":
+            raise ValidationError("--max-alive-period can only be set when --lifecycle-type is 'OnContainerExit'.")
```

Comment:
This is for command `az containerapp sessionpool update`, the client will use `PATCH` to update the sessionpool.
My question is if the customer already set the `lifecycle-type` value to `OnContainerExit` when create the sessionpool or somewhere else. Now She want to update the `max-alive-period`, this validation require customer must input with `--lifecycle-type OnContainerExit`, right?
Generally, we should let the `update` command to update the value if we allow them in the RP.

Reply:
I recommend to check if the value for `lifecycle-type` in the `self.existing_pool_def` is already `OnContainerExit`

Reply:
For this specific case, there is a check of get_argument_lifecycle_type() is not null also in the statement to get the customer's input for `lifecycle_type`. So I think this is just validating that when the customer sets both `lifecycle-type` and `max-alive-period` in this az cli command, the `lifecycle_type` input cannot be `OnContainerExit`.

And I think customer can still only update `max-alive-period`, and `lifecycle-type` should not be mandatory.

But thank you very much for the recommendation, I added some validation using the `existing_pool_def` to validate customer is not providing only the unsupported property for a certain existing `lifecycle_type`.

Also I updated the code style to make the if statement a little clearer